OpenSSL 1.1.1q doesn't build on Mac

Error in our of the unit tests:
```
 test/v3ext.c:201:24: error: implicitly declaring library function 'memcmp' with type 'int (const void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
11:57:39          if (!TEST_true(memcmp(ip1->data, ip2->data, ip1->length) <= 0))
11:57:39                         ^
11:57:39  test/v3ext.c:201:24: note: include the header <string.h> or explicitly provide a declaration for 'memcmp'
11:57:39  1 error generated.
11:57:39  make[1]: *** [test/v3ext.o] Error 1
```
This was [fixed upstream](https://github.com/openssl/openssl/commit/60f011f584d80447e86cae1d1bd3ae24bc13235b) but is not yet part of a release.

--- a/build_darwin_arm64.sh
+++ b/build_darwin_arm64.sh
@@ -27,6 +27,7 @@ cd ../../
 
 tar -xvzf "openssl-$OPENSSL_VERSION.tar.gz" -C arm64
 cd "arm64/openssl-$OPENSSL_VERSION"
+patch -p0 < ../../patch/openssl/v3ext.c.patch
 ./Configure --prefix="$PWD/root" darwin64-arm64-cc no-shared no-dso
 make ${jobs:+-j${jobs}} && make install
 cd ../../

--- a/build_darwin_x86_64.sh
+++ b/build_darwin_x86_64.sh
@@ -27,6 +27,7 @@ cd ../../
 
 tar -xvzf "openssl-$OPENSSL_VERSION.tar.gz" -C x86_64
 cd "x86_64/openssl-$OPENSSL_VERSION"
+patch -p0 < ../../patch/openssl/v3ext.c.patch
 ./Configure --prefix="$PWD/root" darwin64-x86_64-cc no-shared no-dso
 make ${jobs:+-j${jobs}} && make test && make install
 cd ../../

--- a/patch/openssl/v3ext.c.patch
+++ b/patch/openssl/v3ext.c.patch
@@ -1,0 +1,15 @@
+Bug-Brave: https://github.com/brave/tor_build_scripts/issues/101
+Description: Add missing include to unit test to fix Mac builds
+Last-Update: 2022-07-06
+Origin: https://github.com/openssl/openssl/commit/60f011f584d80447e86cae1d1bd3ae24bc13235b
+
+--- test/v3ext.c
++++ test/v3ext.c
+@@ -8,6 +8,7 @@
+  */
+ 
+ #include <stdio.h>
++#include <string.h>
+ #include <openssl/x509.h>
+ #include <openssl/x509v3.h>
+ #include <openssl/pem.h>

